### PR TITLE
Don't allow unwraped variables to be unpacked

### LINKED ISSUE #801: Fatal error by UnwrapArrayMerge and unpack
| Question    | Answer
| ------------| ---------------
| Infection version | 0.14.0
| Test Framework version | PHPUnit  7.4.5
| PHP version | 7.3.9
| Platform    | Ubunut
| Github Repo | -

Php fatal error caused by UnwrapArrayMerge and this diff:  

[M] UnwrapArrayMerge
```diff
--- Original
+++ New
@@ @@
-        return \count($classes) ? array_merge(...$classes) : $classes;
+        return \count($classes) ? ...$classes : $classes;
     }
```

### PR DESCRIPTION
**Pull request by BackEndTea**
This can generate syntax errors

This PR:

- [x] Fixed #801 
- [x] Covered by tests

### CODE REVIEW

File: tests/phpunit/Mutator/Unwrap/UnwrapArrayMergeTest.php [hunk 1]
```diff
@@ -285,6 +285,15 @@ function array_merge($array, $array1, $array2)
 $a = 'array_merge';
 
 $b = $a([1,2,3], [3,4,5]);
+PHP
+            ,
+        ];
+
+        yield 'It does not mutate when the input array is unpacked' => [
```
Comment (maks-rafalko) [inline tests/phpunit/Mutator/Unwrap/UnwrapArrayMergeTest.php]: Could you please also add a test for the following case:

```php
$first = [1, 2, 3, 4, 5];
$other = [[6, 7], [8]];

$result = array_merge($first, ...$other);
```

where only first (`0` index) item should be unwrapped

Comment (BackEndTea) [reply tests/phpunit/Mutator/Unwrap/UnwrapArrayMergeTest.php]: Turns out my original solution didn't allow those, so i changed it to make that decision when mutating instead.

**Comment by maks-rafalko** [comment]
Thank you @BackEndTea

**Comment by BackEndTea** [comment]
Thank you @maks-rafalko

**Pull request merged**